### PR TITLE
Feature/gallery block

### PR DESCRIPTION
**Pull request by Mortadha-abderrhim**
Adds the Gallery Block
galleryBlock

### CODE REVIEW

**Comment by hadiyahya11** [comment]
Everything seems fine to me. Just make sure to refactor some of the code just as Quentin suggested.

File: app/src/androidTest/java/com/pindex/main/home/ExperienceActivityInstrumentedTest.kt [hunk 1]
```diff
@@ -109,6 +109,26 @@ class ExperienceActivityInstrumentedTest {
 
         addBlock(block)
     }
+    /**
```
Comment (lucasrlt) [inline app/src/androidTest/java/com/pindex/main/home/ExperienceActivityInstrumentedTest.kt:112]: Missing space between 2 lines of code
Consider running Code > Reformat Code in Android studio on this file

File: app/src/main/java/com/pindex/main/home/ExperienceActivity.kt [hunk 2]
```diff
@@ -10,7 +10,7 @@ import com.pindex.main.models.BlockDto
 import com.pindex.main.models.ExperienceDto
 import com.pindex.main.ui.fragments.blocks.*
 import com.pindex.main.utils.Constants
-import dagger.hilt.android.AndroidEntryPoint
+
```
Comment (lucasrlt) [inline app/src/main/java/com/pindex/main/home/ExperienceActivity.kt:13]: Unnecessary line break, same comment as before

File: app/src/main/java/com/pindex/main/models/BlockDto.kt [hunk 3]
```diff
@@ -8,6 +8,7 @@ import kotlinx.android.parcel.Parcelize
 @Parcelize
 data class BlockDto(
         var image: ImageBlockDto? = null,
+        var gallery : ArrayList<ImageBlockDto>?=null,
```
Comment (lucasrlt) [inline app/src/main/java/com/pindex/main/models/BlockDto.kt:11]: Missing space before and after the "="

File: app/src/main/java/com/pindex/main/ui/fragments/blocks/GalleryBlockFragment.kt [hunk 4]
```diff
@@ -0,0 +1,92 @@
...
+    private var selectedPosition : Int = 0
+    lateinit var galleryPagerAdapter: GalleryPagerAdapter
+
+
+    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View? {
+        val view = inflater.inflate(R.layout.fragment_pindex_block_gallery,container,false)
+        viewPager = view.findViewById(R.id.pindex_block_gallery_viewPager)
+        galleryPagerAdapter = GalleryPagerAdapter()
+        imagePaths = arguments?.getStringArrayList(Constants.BlockFragmentArguments.GALLERY_BLOCK_IMAGE_PATHS)
+        imageDescriptions = arguments?.getStringArrayList(Constants.BlockFragmentArguments.GALLERY_BLOCK_IMAGE_DESCRIPTIONS)
```
Comment (hadiyahya11) [inline app/src/main/java/com/pindex/main/ui/fragments/blocks/GalleryBlockFragment.kt:36]: Too many lines of code without space between them

File: app/src/main/java/com/pindex/main/ui/fragments/blocks/GalleryBlockFragment.kt [hunk 5]
```diff
@@ -0,0 +1,92 @@
...
+               }
+
+               override fun onPageSelected(position: Int) {
+
+               }
+           }
+
+
+    inner class GalleryPagerAdapter : PagerAdapter(){
+        //Creates a page with the image at the given position in [imagePaths]
```
Comment (lucasrlt) [inline app/src/main/java/com/pindex/main/ui/fragments/blocks/GalleryBlockFragment.kt:63]: Missing space in comment. Consider running Android code formatter on this file too

File: app/src/main/java/com/pindex/main/ui/fragments/blocks/GalleryBlockFragment.kt [hunk 6]
```diff
@@ -0,0 +1,92 @@
...
+        override fun instantiateItem(container: ViewGroup, position: Int): Any {
+            val layoutInflater = activity?.getSystemService(Context.LAYOUT_INFLATER_SERVICE) as LayoutInflater
+            val view= layoutInflater.inflate(R.layout.fragment_pindex_block_gallery_image,container,false)
```
Comment (lucasrlt) [inline app/src/main/java/com/pindex/main/ui/fragments/blocks/GalleryBlockFragment.kt:66]: Missing space "="

File: app/src/main/java/com/pindex/main/utils/Constants.kt [hunk 7]
```diff
@@ -49,6 +49,14 @@ object Constants {
          * The text block text argument.
          */
         const val TEXT_BLOCK_TEXT: String = "text"
+        /**
+         * The Gallery block image paths argument.
```
Comment (lucasrlt) [inline app/src/main/java/com/pindex/main/utils/Constants.kt:53]: Missing spaces, reformat code on this file too

File: app/src/main/java/com/pindex/main/home/ExperienceActivity.kt [hunk 8]
```diff
@@ -69,6 +69,19 @@ class ExperienceActivity : AppCompatActivity() {
...
+               fragment = GalleryBlockFragment()
+               val images: List<Pair<String?,String?>>? = block.gallery?.map { img -> img.imageDescription to img.imagePath }
+
+               fragment.arguments = bundleOf(
+
+                       Constants.BlockFragmentArguments.GALLERY_BLOCK_IMAGE_DESCRIPTIONS to images?.map{img -> img.first},
+                       Constants.BlockFragmentArguments.GALLERY_BLOCK_IMAGE_PATHS to  images?.map{img -> img.second}
+
+
+
```
Comment (hadiyahya11) [inline app/src/main/java/com/pindex/main/home/ExperienceActivity.kt:82]: Maybe run Code > Reformat Code in Android Studio on that file

File: app/src/main/java/com/pindex/main/ui/fragments/blocks/GalleryBlockFragment.kt [hunk 9]
```diff
@@ -0,0 +1,92 @@
...
+import com.pindex.main.R
+import com.pindex.main.utils.Constants
+import com.pindex.main.utils.MediaLoader
+import kotlinx.android.synthetic.main.fragment_pindex_block_gallery_image.view.*
+
+/**
+ * Custom Fragment for the Gallery Block
+ */
+class GalleryBlockFragment: Fragment(R.layout.fragment_pindex_block_gallery) {
+    // image paths' list
```
Comment (lucasrlt) [inline app/src/main/java/com/pindex/main/ui/fragments/blocks/GalleryBlockFragment.kt:22]: Are those comments really necessary ?

File: app/src/main/java/com/pindex/main/ui/fragments/blocks/GalleryBlockFragment.kt [hunk 10]
```diff
@@ -0,0 +1,92 @@
...
+    private var imagePaths: List<String>?=null
+    // image descriptions' list
+    private var imageDescriptions: List<String>?=null
+    lateinit var viewPager: ViewPager
+    private var selectedPosition : Int = 0
+    lateinit var galleryPagerAdapter: GalleryPagerAdapter
+
+
+    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View? {
```
Comment (lucasrlt) [inline app/src/main/java/com/pindex/main/ui/fragments/blocks/GalleryBlockFragment.kt:31]: This function looks quite cluttered, you could maybe add spaces between some lines

**Review by lucasrlt — CHANGES_REQUESTED** [review]
The code is fine, just run Android Studio's Code > Reformat Code tool on some files to maintain code consistencu

**Review by lucasrlt — APPROVED** [review]
The code is consistent and well formatted.
Total coverage decreases a bit but it is still > 80% so it's ok.

**Review by hadiyahya11 — APPROVED** [review]

**Pull request merged**